### DISCUSSION TITLE
Detecting duplicate pairings to the same peer

### DISCUSSION THREAD

**Discussion by bobwith2bees** [WalletConnect/walletconnect-monorepo#2410] [Developer Support]
I asked in the Flutter project, but this is a more general question so posting here.

I was looking for a way to see if my DAPP can detect a duplicate connection to the same wallet so that I can better remediate the confusing configuration.  (Multiple relay sessions to the same app/device.) Is there a recommended way to do this?  Or has the design taken a privacy-forward approach and purposely limited any unique identification points?  

Flutter repo question: https://github.com/WalletConnect/WalletConnectFlutterV2/issues/83
Low priority on this, more curious to see ways to make the UX cleaner.

**Comment by finessevanes** [comment]
Hey, thanks for the question. Are you looking to detect duplicate pairings (as the title suggests) or 
>Multiple relay sessions to the same app/device.

sessions?

**Reply by bobwith2bees** [reply]
Duplicate pairings to the same wallet.  

Do you see a future where a user could connect their desktop and mobile wallets - and my DAPP could ping which is active and send a request to the appropriate session?     If I have multiple pairings  to a single wallet - I might hit the same peer with multiple requests of the same thing.    It also might be an issue for wallets if poorly behaving apps always create a new pairing / forget to send the disconnect etc.   

I know in my app example testing --  I have a few too many pairings.  I just wanted to programatically add a warning for the next developer there's something out of sync.  (would be nice)
